src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,129 @@
+import React, {
+  ChangeEvent,
+  FormEvent,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { Todo, TodoData } from '../../types';
+
+type Props = {
+  todo: Todo,
+  onDelete: () => void,
+  onUpdate: (data: TodoData) => void,
+  beingProcessed?: boolean,
+};
+
+export const TodoItem: React.FC<Props> = React.memo(({
+  todo,
+  onDelete,
+  onUpdate,
+  beingProcessed = false,
+}) => {
+  const { title, completed } = todo;
+  const [beingEdited, setBeingEdited] = useState(false);
+  const [newTitle, setNewTitle] = useState(title);
+  const todoRef = useRef<HTMLDivElement>(null);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const startEditing = useCallback(() => setBeingEdited(true), []);
+  const finishEditing = useCallback(() => setBeingEdited(false), []);
+
+  useEffect(() => {
+    todoRef.current?.addEventListener('dblclick', startEditing);
+
+    return () => todoRef.current?.removeEventListener('dblclick', startEditing);
+  }, []);
+
+  useEffect(() => {
+    if (beingEdited) {
+      inputRef.current?.focus();
+    }
+  }, [beingEdited]);
+
+  const handleToggle = () => {
+    onUpdate({ completed: !todo.completed });
+  };
+
+  const handleInput = useCallback((event: ChangeEvent<HTMLInputElement>) => {
+    setNewTitle(event.target.value.trimStart());
+  }, []);
+
+  const handleRename = (event: FormEvent) => {
+    event.preventDefault();
+
+    if (!newTitle) {
+      onDelete();
+
+      return;
+    }
+
+    if (title !== newTitle) {
+      onUpdate({ title: newTitle });
+    }
+
+    finishEditing();
+  };
+
+  return (
+    <div
+      ref={todoRef}
+      className={classNames(
+        'todo',
+        { completed },
+      )}
+    >
+      <label className="todo__status-label">
+        <input
+          type="checkbox"
+          className="todo__status"
+          checked={completed}
+          onChange={handleToggle}
+        />
+      </label>
+
+      {beingEdited
+        ? (
+          <form
+            onSubmit={handleRename}
+          >
+            <input
+              ref={inputRef}
```

Comment:
use set focus to the input without ref, you can use autoFocus attribute but ref also good

Reply:
I thought about it but linter hates autofocus

Reply:
@nzrii-kysel  https://prnt.sc/rZ1SpEflz-Ko 😢